app/controllers/declaration/SealController.scala

```diff
@@ -42,78 +43,129 @@ class SealController @Inject()(
   errorHandler: ErrorHandler,
   override val exportsCacheService: ExportsCacheService,
   mcc: MessagesControllerComponents,
-  sealPage: seal
+  addPage: seal_add,
+  removePage: seal_remove,
+  summaryPage: seal_summary
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport with ModelCacheable {
 
-  def displayPage(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
-    val declaration = request.cacheModel
-    Ok(sealPage(mode, form(), declaration.seals, declaration.transportDetails.exists(_.container)))
+  def displayAddSeal(mode: Mode, containerId: String): Action[AnyContent] = (authenticate andThen journeyType) {
+    implicit request =>
+      Ok(addPage(mode, Seal.form(), containerId))
   }
 
-  def submitForm(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    val actionTypeOpt = FormAction.bindFromRequest()
-    val boundForm = form().bindFromRequest()
-
-    val data = request.cacheModel
-    val seals = data.seals
-    val hasContainers: Boolean = data.transportDetails.fold(false)(_.container)
-    actionTypeOpt match {
-      case Add                             => addSeal(mode, boundForm, sealsAllowed, seals)
-      case Remove(ids)                     => removeSeal(mode, boundForm, seals, hasContainers, ids)
-      case SaveAndContinue | SaveAndReturn => saveSeal(mode, boundForm, sealsAllowed, seals)
-      case _                               => errorHandler.displayErrorPage()
+  def submitAddSeal(mode: Mode, containerId: String): Action[AnyContent] = (authenticate andThen journeyType).async {
+    implicit request =>
+      val boundForm = Seal.form().bindFromRequest()
+      request.cacheModel.containerData.flatMap(_.containers.find(_.id == containerId)) match {
```

Comment:
It would be worth to add method to `ExportsDeclaration` for finding and modifying containers same as items ?